### DISCUSSION TITLE
Could next/link be used for external sources?

### DISCUSSION THREAD

**Discussion by predaytor** [vercel/next.js#30730] [Help] [answered]
I parse data from CMS to React components. I wonder why next/link should be only used within website navigation. The link can point to both a current website and an external source. I am currently using conditional rendering: 
```ts
const WithNextLink = React.forwardRef<
	React.ElementRef<typeof Link>,
	React.ComponentPropsWithoutRef<typeof Link>
>(({ href, children, ...props }, ref) => {
	const apiHost = new URL(process.env.NEXT_PUBLIC_WORDPRESS_API_URL).host;

	return href.includes(apiHost) ? (
		<NextLink href={new URL(href).pathname} passHref>
			<Link ref={ref} {...props}>
				{children}
			</Link>
		</NextLink>
	) : (
		<Link ref={ref} href={href} {...props}>
			{children}
		</Link>
	);
});
```

**Comment by jamesmosier** [comment, marked as answer]
The purpose of `next/link` is so that Next can perform client side navigation between pages in your website. Next also performs optimizations (e.g. prefetching pages that are linked on your page so they load faster) which is not possible with an external url.

**Reply by predaytor** [reply]
Yes, I know that. I'm just wondering if fallback to the standard behavior of `<a>` could be implemented, if there's external source.
